sklearn/linear_model/_logistic.py

```diff
@@ -2266,21 +2277,29 @@ def score(self, X, y, sample_weight=None, **score_params):
         score : float
             Score of self.predict(X) w.r.t. y.
         """
+        # TODO(1.10): for backwards compatibility, when `sample_weight` becomes a part
+        # of **score_params, it should be skipped in the following check so that people
+        # can still pass it w/o metadata routing enabled.
         _raise_for_params(score_params, self, "score")
+        # TODO(1.10): remove this when sample_weight is removed from the `score`
+        # signature
+        if sample_weight is not None:
+            score_params["sample_weight"] = sample_weight
 
         scoring = self._get_scorer()
         if _routing_enabled():
             routed_params = process_routing(
                 self,
                 "score",
-                sample_weight=sample_weight,
                 **score_params,
             )
         else:
             routed_params = Bunch()
             routed_params.scorer = Bunch(score={})
-            if sample_weight is not None:
-                routed_params.scorer.score["sample_weight"] = sample_weight
+            if score_params.get("sample_weight", None) is not None:
```

Comment:
```suggestion
            if score_params.get("sample_weight") is not None:
```

Reply:
`sample_weight` is not always a part of this dictionary (it's not if the user doesn't pass it), and the `get` here would then raise. So I don't think your change is valid.

Reply:
score_params is a dictionary right? Then `dict.get` returns None by default in Python. The default value that you are assigning is None which is already going to be returned.